make/compiler/Makefile.clang

```diff
@@ -176,7 +176,7 @@ RUNTIME_CXXFLAGS += $(WARN_CXXFLAGS)
 # -Wno-strict-prototypes has to be used because most GASNet prototypes
 # aren't strict.
 #
-CHPL_GASNET_MORE_CFLAGS = -Wno-strict-prototypes -Wno-missing-prototypes -Wno-unused-variable -Wno-unused-function
+CHPL_GASNET_MORE_CFLAGS = -Wno-strict-prototypes -Wno-missing-prototypes -Wno-unused-variable -Wno-unused-function -Wno-cast-function-type-strict
```

Comment:
Do older clangs that we might need to compile Chapel with support this flag `-Wno-cast-function-type-strict`? If not, we need to check the clang version before adding this flag.

Reply:
Added